baseplate/core.py

```diff
@@ -506,6 +528,11 @@ def session(self):
         """:py:class:`~baseplate.core.Session` object for the current context"""
         return Session(id=self._t_request.session.id)
 
+    @cached_property
+    def service(self):
+        """:py:class:`~baseplate.core.Service` object for the current context"""
+        return Service(self.authentication_token)
+
```

Comment:
I'm planning on using this in thing service by first trying to get `request_context.user.id` and if that fails checking `request_context.service.name`. Does that sound reasonable?

Reply:
Sounds OK to start out. If things get more complicated we might want to figure out how to turn the user context into application authorization more natively. I'd really love to hook into Pyramid's authorization stuff where possible.